components/feature/webnotifications/src/main/java/mozilla/components/feature/webnotifications/NativeNotificationBridge.kt

```diff
@@ -37,22 +43,22 @@ internal class NativeNotificationBridge(
         }
 
         with(notification) {
-            loadIcon(iconUrl?.toUri(), origin, Size.DEFAULT)?.let { icon ->
-                builder.setLargeIcon(icon)
+            activityClass?.let {
+                val intent = Intent(context, activityClass)
+                intent.putExtra(EXTRA_ON_CLICK, tag)
+                val pendingIntent = PendingIntent.getActivity(context, requestId, intent, 0)
+                builder.setContentIntent(pendingIntent)
             }
 
-            builder.setContentTitle(title).setContentText(body)
-
-            @Suppress("Deprecation")
-            builder.setVibrate(vibrate)
+            builder.setSmallIcon(launcherIcon).setContentTitle(title).setContentText(text)
```

Comment:
The small icon should be a dedicated monochrome icon instead of just the launcher icon.

Reply:
Yes, but we'll need to get `Notification.Badge` from GV down the road.  However, currently we don't have it.  So I added this to allow user of the feature to pass an icon in.  I'll rename the parameter to `smallIcon` so its more clear.